### DISCUSSION TITLE
Help Compiling to ESM from TypeScript

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [SBoudrias/Inquirer.js#1206] [Q&A]
Hi. I'm having troubles compiling TypeScript projects that use `inquirer`.
All kind of errors. First it was `string_decoder`. Now it's `Error: Dynamic require of "assert" is not supported`.

Every time, I just remove inquirer and the apps compile without issues.

First it was a `Nuxt 3` project. Then a pure TypeScript project with 3 dummy files.

First I used `unbuild` package to compile. Now I'm using `esbuild`
with: `pnpx esbuild ${file}.ts --outfile=.dist/${file}.mjs --platform=node --format=esm --bundle`

I tried using dynamic import to no avail.

Any tip will be appreciated.

**Comment by angelhdzmultimedia** [comment]
### Update
I found a workaround. Instead of leaving the builders (unbuild, esbuild) to pack inquirer in the .dist, I defined a dummy prompt function.
Then, in the built files, I imported inquirer , and replaced the dummy prompt function with the one from inquirer, and it's working.

So my guess is that inquirer depends on packages that are not being converted to ESM...

**Reply by SBoudrias** [reply]
> inquirer depends on packages that are not being converted to ESM

This is very likely with transitive dependencies. FWIW Node.js has no issues dependencies who're common.js - the compatibility issue is a common.js module importing an ESM module. The other way around is supported.

So maybe the build tool used expect something specific; but as long as you export/expose ESM, you won't have an issue with the `inquirer` package.

On other thing you might want to consider is to used the prompts from `@inquirer/*` packages. Those are ESM and common.js compatible.
